### PR TITLE
[MU4] Fix 317951 - Horizontal Frame on system by itself at end of page causes previous system to disappear

### PR DESCRIPTION
**Pull request by njvdberg**
Resolves: https://musescore.org/en/node/317951

Will return the correct bottom y coordinate in case a `SysStaff` contains only a horizontal frame.
In this case the `SysStaff` doens't have a valid skyline and the initial (far too large) maximum value is returned.
This is solved by returning the height of the `SysStaff` in case the skyline is invalid (has no segments).

PR [#7776](https://github.com/musescore/MuseScore/pull/7776) is similar for `3.x`.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/libmscore/skyline.cpp [hunk 1]
```diff
@@ -284,6 +284,12 @@ void SkylineLine::paint(QPainter& p) const
         }
         x1 = x2;
     }
+
```
Comment (Jojo-Schmitz) [inline src/libmscore/skyline.cpp]: still a line too much for my taste ;-)

**Review by Jojo-Schmitz — APPROVED** [review]

File: src/libmscore/skyline.cpp (hunk 1 above)
Comment (njvdberg) [reply src/libmscore/skyline.cpp]: `uncrustify` had the same comment :wink:

Comment (Jojo-Schmitz) [reply src/libmscore/skyline.cpp]: @jojo-bot at work ;-)

Comment (njvdberg) [reply src/libmscore/skyline.cpp]: And I thought giving the bot some extra work :smile:

**Pull request merged**